SIT.Manager/Services/ModService.cs

```diff
@@ -27,23 +28,75 @@ public class ModService(IBarNotificationService barNotificationService,
     private readonly ILocalizationService _localizationService = localizationService;
 
-    public async Task DownloadModsCollection()
+    // Stores the downloaded mods and caches it in the manager's directory
+    private readonly string _localModCache = Path.Combine(AppContext.BaseDirectory, "Mods");
+
+    public string[] RecommendedModInstalls => ["ConfigurationManager"];
+
+    public List<ModInfo> ModList { get; private set; } = [];
+
+    private async Task InstallFiles(string baseSourceDirectory, string baseTargetDirectory, List<string> files)
     {
-        string modsDirectory = Path.Combine(_configService.Config.InstallPath, "SITLauncher", "Mods");
-        if (!Directory.Exists(modsDirectory))
+        foreach (string file in files)
         {
-            Directory.CreateDirectory(modsDirectory);
+            string sourcePath = Path.Combine(baseSourceDirectory, file);
+            string targetPath = Path.Combine(baseTargetDirectory, file);
+            await _filesService.CopyFileAsync(sourcePath, targetPath).ConfigureAwait(false);
         }
+    }
 
-        string[] subDirs = Directory.GetDirectories(modsDirectory);
+    private async Task UninstallFiles(string baseInstallDirectory, List<string> files, string modName)
+    {
+        foreach (string file in files)
+        {
+            string targetPath = Path.Combine(baseInstallDirectory, file);
+            if (File.Exists(targetPath))
+            {
+                File.Delete(targetPath);
+            }
+            else
+            {
+                ContentDialog dialog = new()
+                {
+                    Title = _localizationService.TranslateSource("ModServiceErrorUninstallModTitle"),
+                    Content = _localizationService.TranslateSource("ModServiceErrorUninstallModDescription", modName, file),
+                    CloseButtonText = _localizationService.TranslateSource("ModServiceButtonNo"),
+                    IsPrimaryButtonEnabled = true,
+                    PrimaryButtonText = _localizationService.TranslateSource("ModServiceButtonYes")
+                };
+
+                ContentDialogResult result = await dialog.ShowAsync();
+                if (result != ContentDialogResult.Primary)
+                {
+                    throw new FileNotFoundException(_localizationService.TranslateSource("ModServiceErrorExceptionUninstallModDescription", modName, modName));
+                }
+            }
+        }
+    }
+
+    public void ClearCache()
+    {
+        string[] subDirs = Directory.GetDirectories(_localModCache);
         foreach (string subDir in subDirs)
         {
             Directory.Delete(subDir, true);
         }
-        Directory.CreateDirectory(Path.Combine(modsDirectory, "Extracted"));
+    }
+
+    public async Task DownloadModsCollection()
+    {
+        if (!Directory.Exists(_localModCache))
+        {
+            Directory.CreateDirectory(_localModCache);
+        }
```

Comment:
We can shorten this to just `Directory.CreateDirectory(_localModCache);` as the create directory method doesn't throw on already existing

Reply:
Good point